libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/utils/InternalUtils.kt

```diff
@@ -1,27 +1,49 @@
 package com.mapbox.navigation.core.internal.utils
 
-import com.mapbox.navigation.core.trip.session.MapboxTripSession
+object InternalUtils {
```

Comment:
I don't think we should introduce a new way to set options here. This pattern is up for discussion because it has individually mutable parameters. Also it's unclear why they're `@Volatile`, it's as if they are built for thread safety when the initialization has strict requirements - they need to be initialized when `MapboxNavigation` is constructed or reconstructed.

I think we should simply define a constant

``` kotlin
val nativePollingOptions = PollingConfig(
    LOOK_AHEAD_SECONDS,
    POLLING_PATIENCE_SECONDS,
    POLLING_INTERVAL_SECONDS
)
```

If we want to make these configurable, we should follow the consistent pattern and then make an Options class. If we want that to be internal, change the visibility to `internal`

``` kotlin
internal class StatusPollingOptions private constructor(
  val Double lookAhead,
  val Double unconditionalPatience,
  val Double: unconditionalInterval
) {
  Builder {
    private var Double lookAhead = 1.0
    private var Double unconditionalPatience = 2.0
    private var Double: unconditionalInterval = 2.0

    fun toBuilder(): Builder = Builder().apply { ...

    fun lookAhead(lookAhead: Double) = apply {
        this.lookAhead = lookAhead
    }
    fun unconditionalPatience(unconditionalPatience: Double) = apply {
        this.unconditionalPatience = unconditionalPatience
    }
    fun unconditionalInterval(unconditionalPatience: Double) = apply {
        this.unconditionalInterval = unconditionalInterval 
    }

    fun build(): StatusPollingOptions
...

    private val pollingConfig = PollingConfig(
        statusPollingOptions.lookAhead,
        statusPollingOptions.unconditionalPatience,
        statusPollingOptions.unconditionalInterval
    )
```

if we want to include a level of abstraction from nav native (e.g., milliseconds instead of seconds) Add a mapper. That way we're not adding more logic to `MapboxNavigation` that is untested

``` kotlin
object PollingConfigMapper {
    fun mapFrom(statusPollingOptions: StatusPollingOptions) = PollingConfig(
        lookAhead = statusPollingOptions.lookAheadMillis / ONE_SECOND_IN_MILLIS,
        ...
    )
}
```

Reply:
Ah and see, the old way was making them `@Volatile` because they were watched by different threads
https://github.com/mapbox/mapbox-navigation-android/pull/4419/files#diff-37365d0e69a56e8efddf9ff8e735c04b3474c7df666259edfbc961855c3556a2L543

The new way requires the fields to be created during construction time, so we no longer need them to be `@Volative` because we no longer can dynamically change the polling rates

Reply:
Removed `@Volatile`.